compiler/cpp/src/thrift/generate/t_go_generator.cc

```diff
@@ -424,6 +424,52 @@ bool t_go_generator::is_pointer_field(t_field* tfield, bool in_container_value)
   throw "INVALID TYPE IN type_to_go_type: " + type->get_name();
 }
 
+string t_go_generator::gen_opt_const_values_(t_base_type::t_base tbase, t_const_value *value) {
+  string go_value_str = "&(&struct{x ";
+  switch (tbase) {
+    case t_base_type::TYPE_BOOL:
+      go_value_str += "bool}{";
+      go_value_str += (value->get_integer() > 0 ? "true" : "false");
+      break;
+
+    case t_base_type::TYPE_I8:
+      go_value_str += "int8}{";
+      go_value_str += std::to_string(static_cast<long long>(value->get_integer()));
+      break;
```

Comment:
Three questions:
a) Why are these casts needed?
b) Can we move that whole method next to the other method? 
c) Why do we need two separate methods at all?